src/main/java/net/barrage/school/java/ecatalog/web/MerchantController.java

```diff
@@ -42,6 +41,10 @@ public Merchant getMerchantById(@PathVariable("merchant_id") Long merchantId) {
     @GetMapping("/{merchant_id}/products")
     public List<Product> listProductsByMerchantId(@PathVariable("merchant_id") Long merchantId) {
         var products = productService.listProductsByMerchant(merchantService.getMerchantById(merchantId));
+        Counter counter = Counter.builder("ecatalog.products.listProductsByMerchantId")
```

Comment:
U should not create counter on every endpoint call! It should use singleton pattern - create once and then use in every request the same unique counter.

Reply:
I get it, but what would be solution if i want to measure how many requests each merchant had? Here I add tag with id to every one of them but I am creating new counter on every request which is also incorrect.

Reply:
Theoretically it's a bad metric for monitoring with prometheus/grafana... Imagine u will have 10K merchants. How do you think it will look like? This metric is more about BI, not about health metrics.

Reply:
But the way u use tags is absolutely correct. If u want to avoid creating counter every time, u can cache it in memory. Consider following approach pls - https://www.baeldung.com/guava-memoizer#function-memoization